test/metabase/notification/send_test.clj

```diff
@@ -692,3 +711,41 @@
                  :done?       (fn [cnt] (= cnt (+ 2 queue-size)))
                  :interval-ms 10
                  :timeout-ms  1000})))))
+
+(deftest blocking-queue-test
+  (let [queue (#'notification.send/->BlockingQueue (java.util.concurrent.ArrayBlockingQueue. 10))]
+
+    (testing "put and take operations work correctly"
+      (#'notification.send/put-notification! queue {:id 1 :payload_type :notification/testing :test-value "A"})
+      (is (= {:id 1 :payload_type :notification/testing :test-value "A"}
+             (#'notification.send/take-notification! queue))))
+
+    (testing "multiple notifications are processed in order, no dedup"
+      (#'notification.send/put-notification! queue {:id 1 :payload_type :notification/testing :test-value "A"})
+      (#'notification.send/put-notification! queue {:id 1 :payload_type :notification/testing :test-value "B"})
```

Comment:
Just sanity checking - in practice System Events will never reuse an id, right?

The reason we're not deduping is because it's not needed, not because it's unwanted? 

Not critiquing this test or the need for it.

Reply:
we do have id in system event. This test is just to make sure we use the right queue for system events.

Reply:
I just mean the id is a nanoid so in practice the fact that we don't dedupe is moot, right?